tripy/tripy/frontend/ops/outer.py

```diff
@@ -0,0 +1,54 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from tripy import export, constraints
+import tripy.frontend.utils as frontend_utils
+
+
+@export.public_api(document_under="operations/functions")
+@frontend_utils.convert_inputs_to_tensors(sync_arg_types=[("vec1", "vec2")])
```

Comment:
IMO we generally don't want this - automatic conversion was primarily for operator overloads where it would be unnatural to write, e.g. `a * tp.Tensor(2, dtype=tp.float32)` rather than simply `a * 2`.